pkg/trafficpolicy/trafficpolicy.go

```diff
@@ -102,22 +102,24 @@ func (out *OutboundTrafficPolicy) AddRoute(httpRouteMatch HTTPRouteMatch, weight
 }
 
 // MergeInboundPolicies merges latest InboundTrafficPolicies into a slice of InboundTrafficPolicies that already exists (original)
-// allowPartialHostnamesMatch is set to true when we intend to merge ingress policies with traffic policies
+// allowPartialHostnamesMatch is set to true when we intend to merge ingress policies or traffic policies having host headers
 func MergeInboundPolicies(allowPartialHostnamesMatch bool, original []*InboundTrafficPolicy, latest ...*InboundTrafficPolicy) []*InboundTrafficPolicy {
 	for _, l := range latest {
 		foundHostnames := false
 		for _, or := range original {
+			// Two inbound traffic policies can be merged by comparing their hostnames either partially or completely. A partial match on hostnames should be allowed for the follwoing scenarios :
+			// 1. when an ingress policy is being merged with other inbound traffic policies or
```

Comment:
Ingress should not be merged with other inbound policies?

Reply:
correct, I have updated the comments to reflect this